### ISSUE TITLE
Support for asymmetric keys

### ISSUE THREAD

**Issue by dharezlak** [zalando/spring-cloud-config-aws-kms#82]
This is a question rather than an issue.

When I try to encrypt with an asymmetric KMS key I get the following exception:

```
com.amazonaws.services.kms.model.InvalidKeyUsageException: Algorithm SYMMETRIC_DEFAULT is incompatible with key spec RSA_4096.
```

Is it a limitation of the AWS SDK that only symmetric keys are supported or something could be changed in this project to support asymmetric keys as well?

**Comment by harti2006** [comment]
Thanks for raising this issue. I need to try this out by myself and will get back to you.

**Comment by harti2006** [comment]
Ok, it seems that there is a new property [encryptionAlgorithm](https://docs.aws.amazon.com/kms/latest/APIReference/API_Encrypt.html#KMS-Encrypt-response-EncryptionAlgorithm), that needs to be set in the client in order to support key different than symmetric ones.
This is definitely some useful configuration option to add to this library. I can provide a pull request...

**Comment by harti2006** [comment]
I played around with the asymmetric encryption feature and drafted a pull request. Fell free to try it out in your project.
I just need more time to make the changes backward compatible and gracefully support older versions of the AWS Java SDK.

**Comment by dharezlak** [comment]
Hi, I finally had a chance to test the encryption/decryption with an asymmetric key and it worked just fine. Thank you for supporting this feature!

**Comment by harti2006** [comment]
Glad that it works for you. I definitely need to polish the code and work on tests and documentation. Currently I'm quite busy, so don't expect it to be ready next week ;-)

**Comment by harti2006** [comment]
Was released in version 5.1
